### PR TITLE
DEPR Enforce reversion of Size color aliases

### LINKED ISSUE #49732: BUG: Plotly scatter plots no longer produce color correctly, color alias for c in 1.5
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
import plotly.express as px
df = px.data.iris()

#Works
px.scatter(df, x="sepal_width", y="sepal_length", color="species")

#doesn't work as expected in 1.5.x
df.plot.scatter(x="sepal_width", y="sepal_length", color="species")
```

### Issue Description

Plotly (and perhaps other backends) take a color argument that is a _column name_ to use to make distinct colors for each variable (or even continuous color.) This is very convenient compared to the `c` variable in the pandas backend that takes a list of colors.

In Pandas 1.5, `color` and `c` in the back end code become the same, or [rather `color` is an alias of `c`:](https://github.com/pandas-dev/pandas/blob/e3b0d4d74b1311f6a40ce1ae13c42cc7619178ab/pandas/plotting/_core.py#L1651)

In addition, the code pops out the `color` argument so that it no longer exists in `kwargs` so the color information (expected to be a list of columns) is now absent. Changing the line to

``` 
color = kwargs.get('color')
```

would leave the variable in place, but then other plotting back ends would get color and presumably crash on an expected keyword, and would have to use `c` instead.

I think it would be best to not alias `color` to `c` for scatter plots and let backends like `plotly` continue to work as they did prior to 1.5.x

### Expected Behavior

Plotly `color` kwarg should behave as expected prior to 1.5.x and pass through a column name to the plotly backend, not remove the kwarg and alias it to the `c` kwarg.

### Installed Versions

<details>

/usr/local/opt/miniforge3/lib/python3.10/site-packages/_distutils_hack/__init__.py:33: UserWarning:

Setuptools is replacing distutils.

INSTALLED VERSIONS
------------------
commit           : 91111fd99898d9dcaa6bf6bedb662db4108da6e6
python           : 3.10.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.1.0
Version          : Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:30 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8103
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.1
numpy            : 1.23.4
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 65.5.1
pip              : 22.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.6.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : 2022.5.0
gcsfs            : None
matplotlib       : 3.5.3
numba            : 0.56.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.0
snappy           : None
sqlalchemy       : 1.4.44
tables           : None
tabulate         : 0.8.10
xarray           : None
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : 2022.6
</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #49732 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Docs and whatsnew were added in https://github.com/pandas-dev/pandas/pull/49734, let's enforce it in 2.0.0?

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
yup - 1.5.3 sounds good!

**Comment by phofl** [comment]
Can you update the whatsnew?

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -572,7 +572,7 @@ Removal of prior version deprecations/changes
 - Changed default of ``numeric_only`` to ``False`` in all DataFrame methods with that argument (:issue:`46096`, :issue:`46906`)
 - Changed default of ``numeric_only`` to ``False`` in :meth:`Series.rank` (:issue:`47561`)
 - Enforced deprecation of silently dropping nuisance columns in groupby and resample operations when ``numeric_only=False`` (:issue:`41475`)
-- Enforced reversion of ``color`` as an alias for ``c`` and ``size`` as an alias for ``s`` in function :meth:`DataFrame.plot.scatter` (:issue:`49732`)
+- Changed default of ``numeric_only`` to ``False`` in :meth:`.DataFrameGroupBy.sum` and :meth:`.DataFrameGroupBy.mean` (:issue:`46072`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:575]: This looks like a mistake?

Comment (phofl) [reply doc/source/whatsnew/v2.0.0.rst:575]: Sorry, my mistake.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
cc @MarcoGorelli could you backport manually?

**Comment by phofl** [comment]
@meeseeksdev  backport 1.5.x

**Comment by MarcoGorelli** [comment]
> cc @MarcoGorelli could you backport manually?

yeah will do tomorrow

**Pull request merged**